dll/win32/syssetup/wizard.c

```diff
@@ -2992,6 +2996,28 @@ ProcessSetupInf(
                              (LPBYTE)pSetupData->SourcePath,
                              (wcslen(pSetupData->SourcePath) + 1) * sizeof(WCHAR));
 
+        GetWindowsDirectoryW(szData, ARRAYSIZE(szData));
```

Comment:
I think it would be better to do all this string path construction **just before** that RegCreateKeyExW() call above (line 2971). The rationale being, that in case of failure for constructing those strings required for the "Installation Sources" value, we don't end up with partial/inconsistent registry data (Namely, having the "SourcePath"/"ServicePackSourcePath" strings set, but not the "Installation Sources" one.)

What do you think about that @zecarlos1957 @binarymaster ?